### DISCUSSION TITLE
How can I pass a variable from alpine x-for loop to a Laravel Blade component as a 'props'

### DISCUSSION THREAD

**Discussion by joffreypersia** [alpinejs/alpine#3686] [1. Help] [answered]
Hi, 

Is it possible to pass an Alpine variable from an `x-for` loop inside a blade component ? 

Here is my use case simplified : 
```html
<template x-for="item in items" :key="item.id">
    <p x-text="item.usage.value"></p> // returns the correct value in the DOM
    <x-form.radio-inline :options="$usagesJson" :selectedValue="item.usage.value" /> // doesn't work
</template>
```

I tried : 
```html
<x-form.radio-inline :options="$usagesJson" ::selectedValue="item.usage.value" />
```

```html
<x-form.radio-inline :options="$usagesJson" x-bind:selectedValue="item.usage.value" />
```

```html
<x-form.radio-inline :options="$usagesJson" :selectedValue="x-bind:item.usage.value" />
```

I am using Laravel 9.52.12 and Alpine 3.4.2

Have you an idea? 

Thank you 🙏
Joffrey Persia

**Comment by SimoTod** [comment, marked as answer]
Not possible out of the box. Alpine runs on the browser. Blade runs on the server.
You need to use something like Livewire

**Reply by joffreypersia** [reply]
Thank you

**Comment by hai-shift** [comment]
in 2025 have a solve to this quations?

**Reply by joffreypersia** [reply]
After working with Laravel, Alpine.js, and Livewire for two years, here’s the way you should follow :

**- Static Pages:**
If your page is static, you don’t need to use an Alpine.js x-for. Instead, simply use Blade’s @foreach directive to iterate over your data and render Blade components.

**- Front-End Dynamic Pages (No Server Communication):**
For pages that require client-side interactivity without server communication, Alpine’s loop is sufficient. However, remember that Alpine’s loop doesn’t work with Blade components. This is because Blade components are rendered on the server; the server processes them into HTML blocks, appends them to the page, and only then is the JavaScript initialized.

**- Reactive Pages (With Server Communication):**
If your page needs a reactive loop with server communication, you should opt for Livewire or Inertia.js. For example, create a Livewire component that manages an array of items, and in its view, use a @foreach($items as $item) loop. Whenever the $items array changes, only the necessary parts of the HTML within the Livewire component are re-hydrated.

**- About x-for:**
The x-for directive is essentially an abstraction of a JavaScript loop. However, since it operates purely on the front end—and we’re not using Vue.js or React—JavaScript alone cannot communicate with the server unless you integrate a solution like Inertia.js.
